### ISSUE TITLE
Implement a way to automatically rebuild components whenever typescript code is changed

### ISSUE THREAD

**Issue by DenisTensorWorks** [EpicGames/PixelStreamingInfrastructure#332]
**Component your feature request relates to**
- [x] Build scripts

**Problem your feature solves? Please describe.**
Currently whenever a user makes changes to the typescript code, the user needs to be rebuild that code and potentially any dependent libraries, this can be quite high friction to dev iteration given our monorepo has quite a few packages now.

It would be nice to have a smoother developer iteration cycles, for example, if it could auto detect changes to any of the dependent libraries and rebuild all relevant components, something akin to:

`cd ./Frontend/implementations/typescript`
`npm run watch`

**Comment by mcottontensor** [comment]
Just dumping down this
https://lerna.js.org/
I haven't actually looked into it yet. Just saw mention of it and want to look into it when I get back to this task.

Same deal with https://nx.dev/

**Comment by lukehb** [comment]
Also dumping some idea here:

The high level idea is:

- Each package has its own npm script to start the webpack dev server (that already does watching for us)

- By creating an uber script that starts all the dev servers we should end up in a situation where you can make a change in common and it will rebuild common, all the things that depend on common, and all the things that depend on those things


It would be pretty nice in theory, basically go to repo root and run something like `npm run dev`

Then can just make changes wherever and things will get rebuilt.

**Comment by lukehb** [comment]
Another thought I had that is about speeding up dev iteration:

We should ensure all our packages have a `compile-dev` and `build-dev` option that only does one of the compilation options (`cjs` or `esm` - not both). While compiling both is good for when we go to publish, for dev, we really only want one - doing both at every package in the repo makes the compilation take twice as long.

tldr; cut down iteration time by only doing half the compiling when we know we are devving.

**Comment by mcottontensor** [comment]
The majority of this is in now with the `develop` script for SS. There are some improvements that can and will probably need to be done but they can be new tasks.
